### PR TITLE
Fixes unexpected timeout of http-slowloris

### PR DESCRIPTION
**Pull request by tremblerz**
The default timeout is 30 minutes and can be changed through timelimit argument, but the script opens minimum of 1000 threads for keeping half-http connections hence stopping all the threads as soon as timeout occurs is little difficult but some improvements have been made in this patch so as to increase the precision and make better effort for timing out the concurrent threads fast.

### CODE REVIEW

File: scripts/http-slowloris.nse [hunk 1]
```diff
@@ -62,16 +62,15 @@ portrule = shortport.http
 
 local SendInterval
 local TimeLimit
-
+local end_time
 
 -- this will save the amount of still connected threads
 local ThreadCount = 0
-
+counter = 0
```
Comment (dmiller-nmap) [inline scripts/http-slowloris.nse]: unused global

**Comment by dmiller-nmap** [comment]
I think this is a good approach, but there is a bigger issue with this script: the use of file-scoped variables (treated as "global," though that has a very different meaning for Lua). As I understand it, every instance of this script will share these variables, so if two services are scanned, the second one to start could e.g. prolong the first one's execution time by modifying `end_time`. I think it might have trouble running against a plain HTTP and an HTTPS service at the same time, too, because `Bestopt` would get overwritten.

The better solution would be to limit the number of these variables needed by scoping as narrowly as possible, then passing the ones that really are necessary as keys in the `obj` table (perhaps renamed to `state` or similar) that is currently used only as a condvar key. The goal is that everything is scoped to the `action` function or closer, so that there is not interaction between runs against different hosts and ports.

File: scripts/http-slowloris.nse (hunk 1 above)
Comment (tremblerz) [inline scripts/http-slowloris.nse]: Oh ! I was using that while debugging, will remove it.

**Comment by dmiller-nmap** [comment]
@tremblerz I'm still not sure about the approach of single-threading the http-slowloris script, especially with @TudorComan's investigations into increased parallelism. But these changes are good for improving responsiveness and granularity of the script to timeouts. I tested by setting a timeout of 30s, and the scan time went from 119s before this change to 47s after, which is much closer.

Go ahead and commit this!

**Comment by tudorcoman** [comment]
@dmiller-nmap what are you talking about?

**Comment by dmiller-nmap** [comment]
Lol, sorry @tudorcoman, I meant @Tudor-Coman

**Comment by tremblerz** [comment]
I've got this PR committed into main svn trunk and forgot to specify "closes #402" in commit message so I am closing this PR manually. Thanks for reviewing.

**Pull request closed without merging**